Snowflake.Data/Core/FileTransfer/SFFileTransferAgent.cs

```diff
@@ -185,46 +185,60 @@ class SFFileTransferAgent
         /// </summary>
         public void execute()
         {
-            // Initialize the encryption metadata
-            initEncryptionMaterial();
-
-            if (CommandTypes.UPLOAD == CommandType)
-            {
-                initFileMetadataForUpload();
-            }
-            else if (CommandTypes.DOWNLOAD == CommandType)
+            try
             {
-                initFileMetadata(TransferMetadata.src_locations);
+                // Initialize the encryption metadata
+                initEncryptionMaterial();
 
-                Directory.CreateDirectory(TransferMetadata.localLocation);
-            }
+                if (CommandTypes.UPLOAD == CommandType)
+                {
+                    initFileMetadataForUpload();
+                }
+                else if (CommandTypes.DOWNLOAD == CommandType)
+                {
+                    initFileMetadata(TransferMetadata.src_locations);
 
-            // Update the file metadata with GCS presigned URL
-            updatePresignedUrl();
+                    Directory.CreateDirectory(TransferMetadata.localLocation);
+                }
 
-            foreach (SFFileMetadata fileMetadata in FilesMetas)
-            {
-                // If the file is larger than the threshold, add it to the large files list
-                // Otherwise add it to the small files list
-                if (fileMetadata.srcFileSize > TransferMetadata.threshold)
+                // Update the file metadata with GCS presigned URL
+                updatePresignedUrl();
+
+                foreach (SFFileMetadata fileMetadata in FilesMetas)
                 {
-                    LargeFilesMetas.Add(fileMetadata);
+                    // If the file is larger than the threshold, add it to the large files list
+                    // Otherwise add it to the small files list
+                    if (fileMetadata.srcFileSize > TransferMetadata.threshold)
+                    {
+                        LargeFilesMetas.Add(fileMetadata);
+                    }
+                    else
+                    {
+                        SmallFilesMetas.Add(fileMetadata);
+                    }
                 }
-                else
+
+                // Check command type
+                if (CommandTypes.UPLOAD == CommandType)
                 {
-                    SmallFilesMetas.Add(fileMetadata);
+                    upload();
+                }
+                else if (CommandTypes.DOWNLOAD == CommandType)
+                {
+                    download();
                 }
             }
-
-            // Check command type
-            if (CommandTypes.UPLOAD == CommandType)
+            catch (FileNotFoundException e)
```

Comment:
why do we handle only FileNotFound and IOException? how did we select these 2? what about others?